plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Site Health checks.
+ *
+ * @package optimization-detective
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Adds the Optimization Detective REST API check to site health tests.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param array{direct: array<string, array{label: string, test: string}>}|mixed $tests Site Health Tests.
+ * @return array{direct: array<string, array{label: string, test: string}>} Amended tests.
+ */
+function od_add_rest_api_availability_test( $tests ): array {
+	if ( ! is_array( $tests ) ) {
+		$tests = array();
+	}
+	$tests['direct']['optimization_detective_rest_api'] = array(
+		'label' => __( 'Optimization Detective REST API Endpoint Availability', 'optimization-detective' ),
+		'test'  => static function () {
+			// Note: A closure is used here to improve symbol discovery for the sake of potential refactoring.
+			return od_test_rest_api_availability();
+		},
+	);
+
+	return $tests;
+}
+
+/**
+ * Tests availability of the Optimization Detective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function od_test_rest_api_availability(): array {
+	$response       = od_get_rest_api_health_check_response( false );
+	$result         = od_compose_site_health_result( $response );
+	$is_unavailable = 'good' !== $result['status'];
+	update_option( 'od_rest_api_unavailable', $is_unavailable ? '1' : '0' );
```

Comment:
This intentionally makes the option autoloaded. Maybe we should pass the parameter to make this explicit.

Reply:
See 8926f10844c555b42a770defb377a18392dd422e